www/include/configuration/configObject/contact/formContact.php

```diff
@@ -784,9 +786,14 @@ function myReplace()
 
 if ($form->validate() && $from_list_menu == false) {
     $cctObj = $form->getElement('contact_id');
+    if (!$centreon->user->admin && $contact_id) {
+        $form->getElement('contact_admin')->setValue(0);
```

Comment:
I'm not sure that you should replace the value after the form submit, as you may overwrite values when you shouldn't -> eg : contactA has the right to reach API, but contactB interraction removes the right when the form is submitted.
You may need to hidden these fields when the form is displayed instead of overwritting the value set...